include/deal.II/numerics/data_out_dof_data.templates.h

```diff
@@ -164,19 +188,24 @@ namespace internal
 
                   for (unsigned int j = 0; j < this->finite_elements[i]->size();
                        ++j)
-                    if ((*this->finite_elements[i])[j].reference_cell_type() !=
+                    if ((*this->finite_elements[i])[j].reference_cell_type() ==
                         ReferenceCell::get_hypercube(dim))
-                      quadrature.push_back(*quadrature1);
-                    else
                       quadrature.push_back(*quadrature2);
```

Comment:
Here's a place that lends itself to hard-to-find bugs because quadrature1/2/3 doesn't give away what they correspond to.